Generator/Sources/NeedleFramework/Parsing/Pluginized/Tasks/PluginizedASTDeclarationParserTask.swift

```diff
@@ -39,97 +39,117 @@ class PluginizedDeclarationsParserTask: AbstractTask<PluginizedDependencyGraphNo
-        let (pluginizedComponents, nonCoreComponents, pluginExtensions) = try parsePluginizedStructures()
+        let visitor = Visitor(sourceHash: ast.sourceHash)
+        visitor.walk(ast.sourceFileSyntax)
+        let pluginizedComponents = visitor.pluginizedComponents
+        let nonCoreComponents = visitor.nonCoreComponents
+        let pluginExtensions = visitor.pluginExtensions
+        
         return PluginizedDependencyGraphNode(pluginizedComponents: pluginizedComponents, nonCoreComponents: nonCoreComponents, pluginExtensions: pluginExtensions, components: baseNode.components, dependencies: baseNode.dependencies, imports: baseNode.imports)
     }
 
     // MARK: - Private
 
     private let ast: AST
+}
 
-    private func parsePluginizedStructures() throws -> ([PluginizedASTComponent], [ASTComponent], [PluginExtension]) {
-        var pluginizedComponents = [PluginizedASTComponent]()
-        var nonCoreComponents = [ASTComponent]()
-        var pluginExtensions = [PluginExtension]()
-
-        let substructures = ast.structure.substructures
-        for substructure in substructures {
-            if substructure.isPluginizedComponent {
-                let (dependencyProtocolName, pluginExtensionName, nonCoreComponentName) = try substructure.pluginizedGenerics()
-                let properties = try substructure.properties()
-                // Pluginized components are never root.
-                let component = ASTComponent(name: substructure.name, dependencyProtocolName: dependencyProtocolName, isRoot: false, sourceHash: ast.sourceHash, properties: properties, expressionCallTypeNames: substructure.uniqueExpressionCallNames)
-                pluginizedComponents.append(PluginizedASTComponent(data: component, pluginExtensionType: pluginExtensionName, nonCoreComponentType: nonCoreComponentName))
-            } else if substructure.isNonCoreComponent {
-                let dependencyProtocolName = try substructure.dependencyProtocolName(for: "NonCoreComponent")
-                let properties = try substructure.properties()
-                // Non-core components are never root.
-                let component = ASTComponent(name: substructure.name, dependencyProtocolName: dependencyProtocolName, isRoot: false, sourceHash: ast.sourceHash, properties: properties, expressionCallTypeNames: substructure.uniqueExpressionCallNames)
-                nonCoreComponents.append(component)
-            } else if substructure.isPluginExtension {
-                let properties = try substructure.properties()
-                pluginExtensions.append(PluginExtension(name: substructure.name, properties: properties))
-            }
+private final class Visitor: BaseVisitor {
+    private(set) var pluginExtensions: [PluginExtension] = []
+    private(set) var pluginizedComponents: [PluginizedASTComponent] = []
+    private(set) var nonCoreComponents: [ASTComponent] = []
+    
+    private var currentPluginizedComponentNode: EntityNode?
```

Comment:
nit: since we only care about the name, we could make this `currentPluginizedComponentName` and a `String`